### ISSUE TITLE
Possible Non-deterministic behaviour

### ISSUE THREAD

**Issue by DRealHammer** [ukoethe/bomberman_rl#16]
If two agents are next to each other and move at the same time in the same direction that is blocked by one of the agents, sometimes only the not blocked one and sometimes also the blocked one can move.

I assume it is related to how the game updates the players (random order?) or maybe just how the user input works.

Reproduce: create a game with two user_agent player (coin_heaven is the easiest setting), move one agent to the other,  try moving in a direction, that is blocked by one agent multiple times

**Comment by fdraxler** [comment]
You are right that this is non-deterministic. However, randomness is not a problem per se for reinforcement learning, but you can think about it as a random part in the game. A good agent will probably not rely on an action which only succeeds in few cases, unless it is very effective in the cases in which it works. Can you explain why randomness is a problem and do you have a suggestion how the game can be changed to counteract this?
